lib/order.rb

```diff
@@ -0,0 +1,38 @@
+class Order
+
+  attr_accessor :current_order, :total
+
+  def initialize(menu)
+    @current_order = {}
+    @total = 0
+    @dishes = menu.dishes
+    @sum = 0
+  end
+
+  def add_to_order(dish, quantity)
+    raise 'Dish not available: not part of the menu' unless @dishes.include?(dish)
+    @current_order.merge!({ dish => quantity })
+    calculate_total
+  end
+
+  def complete_order(sum)
+    @sum = sum
+    raise 'Payment error: incorrect payment submitted' unless sum == @total
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)